### PR TITLE
Merge RStudio image into master, update all bases

### LINKED ISSUE #28: Pin nodejs version
Especially for the python3-minimal base, if we want to install nodejs, it will then require a rebuild for the jupyterlab extensions if the nodejs version doesn't match (which it won't now if you install latest 10.x).

If we pin this, then it's easier to install the same nodejs version. Maybe down the road we even have this as a single click option in the environment?

### PR DESCRIPTION
**Pull request by davclark**
Changes overall are pretty small. Unfortunately, the diff between revisions is a little annoying to look at. You may want to check out the branch and do diffs on previous revisions. Note that I also made *tiny* changes to some previous revisions because the information was wrong. In retrospect, this will affect almost no-one.

Note that we started including Pillow 5.4 for now, as Pillow 6.0 is broken and is installed by default by libraries like PyTorch.

We also close #28, pinning nodejs to the latest LTS release available via conda.

Also - after discussion with Dean, I killed the CUDA 8.0 base. If folks ask for it, it's SUPER fast to deploy. We may make a separate "extended" repo for less well-tested bases, so we can focus on a highly useful core.

Please bang on the bases via the latest edge build by including the following in `~/gigantum/.labmanager/config.yaml`:

```
environment:
  repo_url:
    - "https://github.com/gigantum/base-images.git@rserver"
```

### CODE REVIEW

File: _templates/python3-minimal-template/Dockerfile [hunk 1]
```diff
@@ -80,4 +87,4 @@ EXPOSE 8888
 WORKDIR $GIGANTUM_WORKDIR
 
 # CMD may be updated during final image assembly, but this keeps a container alive.
-CMD ["tail -f /dev/null"]
+CMD ["tail", "-f", "/dev/null"]
```
Comment (davclark) [inline _templates/python3-minimal-template/Dockerfile]: This has been broken for who knows how long

Comment (dkleissa) [reply _templates/python3-minimal-template/Dockerfile]: crazy...i guess it still worked ok or else the container would close

File: python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  tag: 8f2a9ba872-2019-05-22
+languages:
+  - python3
+development_tools:
+  - jupyterlab
+  - notebook
+package_managers:
+  - pip:
+  - conda3:
+    - pillow=5.4*
```
Comment (dkleissa) [inline python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml]: why is pillow listed here? Built from python3 minimal, so should be "empty, right?

File: _templates/python3-minimal-template/Dockerfile [hunk 3]
```diff
@@ -50,17 +50,24 @@ RUN cd /tmp && \
...
     rm Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh && \
     conda config --system --prepend channels conda-forge && \
+    # We don't want the `r` channel
+    conda config --system --remove channels defaults && \
+    conda config --system --append channels anaconda && \
     conda config --system --set auto_update_conda false && \
     conda config --system --set show_channel_urls true && \
     # We update conda explicitly to get any gains in performance, etc.
     conda update --quiet --yes conda && \
     conda update --all --quiet --yes && \
```
Comment (dkleissa) [inline _templates/python3-minimal-template/Dockerfile]: even though we pin conda, we run update on it? Probably fine, but maybe unnecessary?

File: python2-minimal/Dockerfile [hunk 4]
```diff
@@ -45,12 +45,16 @@ RUN cd /tmp && wget --quiet https://repo.continuum.io/miniconda/Miniconda2-${MIN
...
+    conda config --system --remove channels defaults && \
+    conda config --system --append channels anaconda && \
     conda config --system --set auto_update_conda false && \
     conda config --system --set show_channel_urls true && \
     conda update --quiet --yes conda && \
     conda update --all --quiet --yes && \
     # notebook 5.3 through 5.7.1 have security holes
-    conda install --quiet --yes -c conda-forge 'jupyterlab=0.33' 'notebook>=5.7.2,<5.8' && \
+    # jupyterlab 0.33 is the final version that supports python 2.7
+    conda install --quiet --yes -c conda-forge 'jupyterlab=0.33' 'notebook>=5.7.2,<5.8' pillow=5.4 && \
```
Comment (dkleissa) [inline python2-minimal/Dockerfile]: why are we installing pillow into the minimal? If worried about pytorch, shouldn't this be just in the CUDA bases?

File: python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml (hunk 2 above)
Comment (dkleissa) [reply python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml]: nevermind...makes sense for CUDA base...just maybe not minimal?

File: _templates/python3-minimal-template/Dockerfile (hunk 3 above)
Comment (davclark) [reply _templates/python3-minimal-template/Dockerfile]: I think the update on conda itself was due to the hope-against-hope that maybe a later version will improve performance. Updating the rest of the packages, I dunno. I agree that a miniconda release is likely to be more coherently tested for the core configuration of packages (and the conda update should bring along anything that's critical for it).

Question is, do you want to make an issue for the next batch of bases, or re-generate all of them?

File: python2-minimal/Dockerfile (hunk 4 above)
Comment (davclark) [reply python2-minimal/Dockerfile]: I agree... I was just going for parity between python3- and python2-minimal, but the reason I give for python3-minimal doesn't hold for python2. I'm happy to rebuild this one. It's only like 11MB, IIRC.

File: python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml (hunk 2 above)
Comment (davclark) [reply python3-minimal-cuda-10.0/python3-minimal-cuda-10.0_r1.yaml]: The issue is that python3-minimal is built from the same template as the CUDA bases. So, it adds complexity to make them different - both for us and the user.

Right now, the description is `python3-minimal` built on a CUDA-enabled ubuntu base. I'd prefer not to add exceptions to that.

File: python2-minimal/Dockerfile (hunk 4 above)
Comment (davclark) [reply python2-minimal/Dockerfile]: For your complete edification (run in python2-minimal from this PR)

```
root@3b33fdc38c44:/opt/conda/pkgs# du -d 1 -h
13M	./conda-4.6.14-py27_0
292K	./libffi-3.2.1-he1b5a44_1006
2.8M	./pyzmq-18.0.1-py27hc4ba49a_1
256K	./configparser-3.7.3-py27_1
100K	./simplegeneric-0.8.1-py_1
724K	./mistune-0.8.4-py27h14c3975_1000
828K	./libedit-3.1.20170329-hf8c457e_1001
4.2M	./cryptography-2.6.1-py27h72c5cf5_0
412K	./enum34-1.1.6-py27_1001
2.7M	./pillow-5.4.1-py27h00a061d_1000
736K	./idna-2.8-py27_1000
384K	./certifi-2019.3.9-py27_0
3.0M	./freetype-2.10.0-he983fc9_0
37M	./notebook-5.7.8-py27_0
15M	./openssl-1.1.1b-h14c3975_1
556K	./pyopenssl-19.0.0-py27_0
188K	./scandir-1.10.0-py27h14c3975_0
144K	./singledispatch-3.4.0.3-py27_1000
76K	./backports-1.0-py_2
1.2M	./prompt_toolkit-1.0.15-py_1
144K	./ptyprocess-0.6.0-py_1001
1.1M	./jpeg-9c-h14c3975_1001
48M	./jupyterlab-0.33.12-py27_0
140K	./webencodings-0.5.1-py_1
588K	./jinja2-2.10.1-py_0
192K	./jupyterlab_launcher-0.11.2-py_0
500K	./pexpect-4.7.0-py27_0
3.4M	./zeromq-4.3.1-hf484d3e_1000
288K	./wheel-0.33.4-py27_0
504K	./pycosat-0.6.3-py27h14c3975_1001
572K	./pyrsistent-0.15.2-py27h516909a_0
196K	./ipython_genutils-0.2.0-py_1
356K	./ca-certificates-2019.3.9-hecc5488_0
46M	./python-2.7.15-h721da81_1008
500K	./jupyter_client-5.2.4-py_3
476K	./jupyter_core-4.4.0-py_0
9.4M	./pip-19.1-py27_0
13M	./ncurses-6.1-hf484d3e_1002
3.9M	./tornado-5.1.1-py27h14c3975_1000
1.6M	./xz-5.2.4-h14c3975_1001
12M	./tk-8.6.9-h84994c4_1001
208K	./olefile-0.46-py_0
116K	./backports.shutil_get_terminal_size-1.0.0-py_3
528K	./requests-2.22.0-py27_0
9.4M	./libstdcxx-ng-8.2.0-hdf63c60_1
216K	./pathlib2-2.3.3-py27_1000
260K	./attrs-19.1.0-py_0
2.3M	./libtiff-4.0.10-h648cc4a_1001
1.1M	./ipykernel-4.10.0-py27_1
184K	./pysocks-1.7.0-py27_0
224K	./futures-3.2.0-py27_1000
160K	./wcwidth-0.1.7-py_1
5.2M	./sqlite-3.26.0-h67949de_1001
172K	./ipaddress-1.0.22-py_1
292K	./prometheus_client-0.6.0-py_0
224K	./terminado-0.8.2-py27_0
3.5M	./setuptools-41.0.1-py27_0
160K	./six-1.12.0-py27_1000
96K	./backports_abc-0.5-py_1
420K	./yaml-0.1.7-h14c3975_1001
1.2M	./pycparser-2.19-py27_1
1.9M	./readline-7.0-hf8c457e_1001
132K	./entrypoints-0.3-py27_1000
1.1M	./cffi-1.12.3-py27h8022711_0
25M	./libgcc-ng-8.2.0-hdf63c60_1
596K	./python-dateutil-2.8.0-py_0
120K	./decorator-4.4.0-py_0
112K	./pandocfilters-1.4.2-py_1
1.2M	./chardet-3.0.4-py27_1003
1.4M	./ruamel_yaml-0.15.71-py27h14c3975_1000
1.9M	./nbconvert-5.5.0-py_0
6.2M	./ipython-5.8.0-py27_0
180K	./defusedxml-0.5.0-py_1
1.4M	./libpng-1.6.37-hed695b0_0
3.7M	./pygments-2.4.0-py_0
127M	./pandoc-2.7.2-0
124K	./pickleshare-0.7.5-py27_1000
680K	./jsonschema-3.0.1-py27_0
196K	./markupsafe-1.1.1-py27h14c3975_0
140K	./send2trash-1.5.0-py_0
440K	./zlib-1.2.11-h14c3975_1004
1.2M	./asn1crypto-0.24.0-py27_1003
912K	./traitlets-4.3.2-py27_1000
268K	./subprocess32-3.5.4-py27h516909a_0
672K	./nbformat-4.4.0-py_1
944K	./urllib3-1.24.3-py27_0
268K	./testpath-0.4.2-py_1001
1.5M	./libsodium-1.0.16-h14c3975_1001
144K	./functools32-3.2.3.2-py_3
856K	./bleach-3.1.0-py_0
426M	.
```

Comment (davclark) [reply python2-minimal/Dockerfile]: pandoc has [come up for similar reasons on StackOverflow](https://stackoverflow.com/questions/48931163/conda-forge-nbconvert-package-without-pandoc)... we could remove it with `--force`:

**Comment by davclark** [comment]
@dkleissa or @billvb please merge this when you release the next version of the Client? It should be merged before the release announcement goes out.

I apologize for breaking DCO - there were a few commits from AGES ago when I was setting up a new OS install.

**Pull request merged**